### ISSUE TITLE
[Bug]: Intel GPU Monitoring not working

### ISSUE THREAD

**Issue by howetech** [netdata/netdata#17400]
### Bug description

Intel GPU Monitoring is not working with the intelgpu plugin.  

When debugging with go.d.plugin -d -m intelgpu the following errors are reported:

ERR intelgpu/intelgpu.go:59 process exited before the first sample was collected collector=intelgpu job=intelgpu
ERR module/job.go:231 init failed collector=intelgpu job=intelgpu
CONFIG go.d:collector:intelgpu:intelgpu status failed

Manually running ./ndsudo igt-json --interval 1 as the netdata uesr returns:

Failed to initialize PMU! (Permission denied)

### Expected behavior

Expecting this plugin to return the metrics as per this guide:

https://learn.netdata.cloud/docs/collecting-metrics/hardware-devices-and-sensors/intel-gpu

### Steps to reproduce

1.Run the latest nightly build
2. no intel gpu metrics reported
3. running debug command results in errors as per above
...

### Installation method

kickstart.sh

### System info

```shell
Linux storage 6.5.13-1-pve #1 SMP PREEMPT_DYNAMIC PMX 6.5.13-1 (2024-02-05T13:50Z) x86_64 GNU/Linux
/etc/os-release:PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
/etc/os-release:NAME="Debian GNU/Linux"
/etc/os-release:VERSION_ID="12"
/etc/os-release:VERSION="12 (bookworm)"
/etc/os-release:VERSION_CODENAME=bookworm
/etc/os-release:ID=debian
```

### Netdata build info

```shell
Packaging:
    Netdata Version ____________________________________________ : v1.45.0-177-nightly
    Installation Type __________________________________________ : binpkg-deb
    Package Architecture _______________________________________ : x86_64
    Package Distro _____________________________________________ :
    Configure Options __________________________________________ : dummy-configure-command
Default Directories:
    User Configurations ________________________________________ : /etc/netdata
    Stock Configurations _______________________________________ : /usr/lib/netdata/conf.d
    Ephemeral Databases (metrics data, metadata) _______________ : /var/cache/netdata
    Permanent Databases ________________________________________ : /var/lib/netdata
    Plugins ____________________________________________________ : /usr/libexec/netdata/plugins.d
    Static Web Files ___________________________________________ : /var/lib/netdata/www
    Log Files __________________________________________________ : /var/log/netdata
    Lock Files _________________________________________________ : /var/lib/netdata/lock
    Home _______________________________________________________ : /var/lib/netdata
Operating System:
    Kernel _____________________________________________________ : Linux
    Kernel Version _____________________________________________ : 6.5.13-1-pve
    Operating System ___________________________________________ : Debian GNU/Linux
    Operating System ID ________________________________________ : debian
    Operating System ID Like ___________________________________ : unknown
    Operating System Version ___________________________________ : 12 (bookworm)
    Operating System Version ID ________________________________ : none
    Detection __________________________________________________ : /etc/os-release
Hardware:
    CPU Cores __________________________________________________ : 8
    CPU Frequency ______________________________________________ : 3300000000
    RAM Bytes __________________________________________________ : 67092598784
    Disk Capacity ______________________________________________ : 111401961578496
    CPU Architecture ___________________________________________ : x86_64
    Virtualization Technology __________________________________ : none
    Virtualization Detection ___________________________________ : systemd-detect-virt
Container:
    Container __________________________________________________ : none
    Container Detection ________________________________________ : systemd-detect-virt
    Container Orchestrator _____________________________________ : none
    Container Operating System _________________________________ : none
    Container Operating System ID ______________________________ : none
    Container Operating System ID Like _________________________ : none
    Container Operating System Version _________________________ : none
    Container Operating System Version ID ______________________ : none
    Container Operating System Detection _______________________ : none
Features:
    Built For __________________________________________________ : Linux
    Netdata Cloud ______________________________________________ : YES
    Health (trigger alerts and send notifications) _____________ : YES
    Streaming (stream metrics to parent Netdata servers) _______ : YES
    Back-filling (of higher database tiers) ____________________ : YES
    Replication (fill the gaps of parent Netdata servers) ______ : YES
    Streaming and Replication Compression ______________________ : YES (zstd lz4 gzip)
    Contexts (index all active and archived metrics) ___________ : YES
    Tiering (multiple dbs with different metrics resolution) ___ : YES (5)
    Machine Learning ___________________________________________ : YES
Database Engines:
    dbengine (compression) _____________________________________ : YES (zstd lz4)
    alloc ______________________________________________________ : YES
    ram ________________________________________________________ : YES
    none _______________________________________________________ : YES
Connectivity Capabilities:
    ACLK (Agent-Cloud Link: MQTT over WebSockets over TLS) _____ : YES
    static (Netdata internal web server) _______________________ : YES
    h2o (web server) ___________________________________________ : YES
    WebRTC (experimental) ______________________________________ : NO
    Native HTTPS (TLS Support) _________________________________ : YES
    TLS Host Verification ______________________________________ : YES
Libraries:
    LZ4 (extremely fast lossless compression algorithm) ________ : YES
    ZSTD (fast, lossless compression algorithm) ________________ : YES
    zlib (lossless data-compression library) ___________________ : YES
    Brotli (generic-purpose lossless compression algorithm) ____ : NO
    protobuf (platform-neutral data serialization protocol) ____ : YES (system)
    OpenSSL (cryptography) _____________________________________ : YES
    libdatachannel (stand-alone WebRTC data channels) __________ : NO
    JSON-C (lightweight JSON manipulation) _____________________ : YES
    libcap (Linux capabilities system operations) ______________ : NO
    libcrypto (cryptographic functions) ________________________ : YES
    libyaml (library for parsing and emitting YAML) ____________ : YES
Plugins:
    apps (monitor processes) ___________________________________ : YES
    cgroups (monitor containers and VMs) _______________________ : YES
    cgroup-network (associate interfaces to CGROUPS) ___________ : YES
    proc (monitor Linux systems) _______________________________ : YES
    tc (monitor Linux network QoS) _____________________________ : YES
    diskspace (monitor Linux mount points) _____________________ : YES
    freebsd (monitor FreeBSD systems) __________________________ : NO
    macos (monitor MacOS systems) ______________________________ : NO
    statsd (collect custom application metrics) ________________ : YES
    timex (check system clock synchronization) _________________ : YES
    idlejitter (check system latency and jitter) _______________ : YES
    bash (support shell data collection jobs - charts.d) _______ : YES
    debugfs (kernel debugging metrics) _________________________ : YES
    cups (monitor printers and print jobs) _____________________ : YES
    ebpf (monitor system calls) ________________________________ : YES
    freeipmi (monitor enterprise server H/W) ___________________ : YES
    nfacct (gather netfilter accounting) _______________________ : YES
    perf (collect kernel performance events) ___________________ : YES
    slabinfo (monitor kernel object caching) ___________________ : YES
    Xen ________________________________________________________ : YES
    Xen VBD Error Tracking _____________________________________ : NO
    Logs Management ____________________________________________ : YES
Exporters:
    AWS Kinesis ________________________________________________ : NO
    GCP PubSub _________________________________________________ : NO
    MongoDB ____________________________________________________ : YES
    Prometheus (OpenMetrics) Exporter __________________________ : YES
    Prometheus Remote Write ____________________________________ : YES
    Graphite ___________________________________________________ : YES
    Graphite HTTP / HTTPS ______________________________________ : YES
    JSON _______________________________________________________ : YES
    JSON HTTP / HTTPS __________________________________________ : YES
    OpenTSDB ___________________________________________________ : YES
    OpenTSDB HTTP / HTTPS ______________________________________ : YES
    All Metrics API ____________________________________________ : YES
    Shell (use metrics in shell scripts) _______________________ : YES
Debug/Developer Features:
    Trace All Netdata Allocations (with charts) ________________ : NO
    Developer Mode (more runtime checks, slower) _______________ : NO
```

**Comment by ilyam8** [comment]
Hi, @howetech. Unfortunately, I can't be of much help as I don't have a server with an Intel iGPU.

I asked @stelfrag to test nightly Netdata on his laptop (Ubuntu with Intel iGPU). He tried the following install methods:
1. from source
2. native package
3. docker

They all worked: 1-2 out of the box, 3 (docker) requires additional setup.

It is known that [intel_gpu_top must be run as root or have CAP_PERFMON](https://gitlab.freedesktop.org/drm/igt-gpu-tools/-/blob/master/tools/intel_gpu_top.c#L2757). But using `ndsudo` should eliminate this requirement - and it does in our tests.

- double-check that `ndsudo` has setuid bit and is owned by root:netdata.
- try adding CAP_PERFMON to `intel_gpu_top`.

**Comment by howetech** [comment]
Turns out the issue was with CAP_PERFMON.  I previously applied this to intel_gpu_top trying to get this plugin to work (and as per previous instructions).  

I reinstalled intel_gpu_top without applying CAP_PERFMON and the plugin is now working perfectly.

Thanks.

**Comment by ilyam8** [comment]
Oh interesting, so you get "Permission denied" if you add  CAP_PERFMON to intel_gpu_top? 🤔 Didn't know that. @stelfrag if you have time please test this scenario.

**Comment by stelfrag** [comment]
> Oh interesting, so you get "Permission denied" if you add CAP_PERFMON to intel_gpu_top? 🤔 Didn't know that. @stelfrag if you have time please test this scenario.

Still works :man_shrugging:

**Comment by edlins** [comment]
I was experiencing disappearing Intel iGPU metrics as well and spent some time testing it.
I found that if I edited the `/etc/netdata/netdata.conf` and set an `update every` either in `[global]` or in `[db]` that the metrics disappeared.  Removing all `update every` restored the metrics.
It seems to be a bug in changing the update rate from the global config.  This should be pretty easy to reproduce.
Used `kickstart.sh --nightly-channel`.
